components/skin/ui/src/main/resources/PhenoTips/CreatePatientRecordForm.xml

```diff
@@ -50,28 +50,40 @@ $xwiki.jsx.use('PhenoTips.Widgets')##
     #if ($xwiki.exists('PhenoTips.StudyBindingClass'))
       ## Is there more than one sheet option?
       ## Check out all the groups this user is part of
-      #set ($toProcess = $collectionstool.queue)
-      #set ($discard = $toProcess.add($xcontext.userReference.toString()))
-      #set ($groups = [])
-      #set ($studies = [])
-      #foreach ($iteration in [0..1000])
-        #if ($toProcess.empty)
-          #break
+      #set($studiesVisibilityPreference=$xwiki.getDocument('PhenoTips.StudyClass').getObject('PhenoTips.StudyVisibilityClass').getProperty('visibility').value)
+      #if ($studiesVisibilityPreference == 'open')
+        #set ($studyClassName = 'PhenoTips.StudyClass')
+        #set ($studyTemplateName = 'PhenoTips.StudyTemplate')
+        #set ($query = $services.query.xwql("from doc.object(${studyClassName}) as std where doc.fullName &lt;&gt; :template").bindValue('template', $studyTemplateName))
+        #set ($studyDocs = $xwiki.wrapDocs($query.execute()))
+        #set ($studies = $collectionstool.sortedSet)
+        #foreach ($study in $studyDocs)
+          #set ($discard = $studies.add("$study"))
```

Comment:
It's safer to use:

    #set ($discard = $studies.add($study.documentReference.toString()))

It's safest to use:

    #set ($discard = $studies.add($services.model.serialize($study.documentReference, 'default')))